### DISCUSSION TITLE
CharacterState.Falling이 다양한 상황에서 다양한 한계와 버그를 발생시킴.

### DISCUSSION THREAD

**Discussion by mzdokebi** [naverz/zepeto-studio-kor#916] [5. Q&A]
떨어지는 상태에는 텔레포트도 안되고 제스쳐도 안됩니다. 왜 막아뒀는지 일단 이해도 안되지만 이것 때문에 다양한 상황에서 너무나 많은 제약 사항이 생깁니다. 특정 구간에 진입하면 트리거를 발생시키고 탈것에 탄다든지 수영을 한다든지 하는 상황을 만들고 싶은데 점프로 이 구간에 진입하면 CharacterState.Falling으로 인해서 전부 캔슬 됩니다.

**Comment by mzdokebi** [comment]
트리거 구간에 진입시 수영을 한다고 가정. 스크립트를 범용으로 사용하기 위해서 캐릭터의 동작은 제스쳐로 고정하고 캐릭터를 더미에 붙이고 더미를 움직인다. 하이드된 더미에 수영 동작이면 수영이 되는 것이고 보이는 서핑보드에 서핑 동작이면 서핑이 된다. 점프로 트리거 구간 진입시 폴링 동작으로 인해서 제스쳐는 취소되고 일반적인 조작 상황으로 돌아간다. 임시로 트리거 구간에 캐릭터를 조작하는 높이와 비슷하게 Collider를 배치해서 캐릭터가 떨어지는 것을 막고 이벤트를 다시 발생시킨다. 좀 어색하지만 이걸로 해결은 된다. Collider의 높이는 어느 정도가 좋은지는 테스트가 부족해서 모르겠다. 각자 알아서~

**Comment by Zepeto-TechEvangelist** [comment]
월드 1.6.0 버전에서 수정 완료되었습니다 :)
